akka-http-core/src/main/scala/akka/http/impl/model/parser/CommonRules.scala

```diff
@@ -78,7 +78,9 @@ private[parser] trait CommonRules { this: Parser with StringBuilding ⇒
   def `day-name` = rule(
     "Sun" ~ push(0) | "Mon" ~ push(1) | "Tue" ~ push(2) | "Wed" ~ push(3) | "Thu" ~ push(4) | "Fri" ~ push(5) | "Sat" ~ push(6))
 
-  def date1 = rule { day ~ ' ' ~ month ~ ' ' ~ year }
+  def `date-sep` = rule { " " | "-" }
```

Comment:
For rules that simply test membership in a certain set of chars it's most efficient if you simply add a new `CharPredicate` instance to `akka.http.impl.model.parser.CharacterClasses` instead of solving it via a parser rule.